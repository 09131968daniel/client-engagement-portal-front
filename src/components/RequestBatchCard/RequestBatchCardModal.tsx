import React from 'react'
import classes from './RequestBatchCardModal.module.scss'

/**
 * @function RequestBatchCardModal
 * Component to display when the client requests a new batch.
 */
const RequestBatchCardModal = () => {
    return (
        <div className={classes.RequestBatchCardModal}>
            <div>Request Sent Successfully</div>
            <div className={classes.XButton}>X</div>
        </div>
    )
}

export default RequestBatchCardModal;
