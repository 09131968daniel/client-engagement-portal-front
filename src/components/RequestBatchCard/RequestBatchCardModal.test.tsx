import React from 'react'
import { shallow } from 'enzyme'
import RequestBatchCardModal from './RequestBatchCardModal'

test('Component Should Render', ()=> {
    const component = shallow(<RequestBatchCardModal />);
    expect(component.render()).toBeTruthy();
})

test('Button Text is \'X\'', ()=>{
    
    const component = shallow(<RequestBatchCardModal />);
    
    expect(component.text().includes('X')).toBe(true);
    
});

test('Modal Text is \'Request Sent Successfully\'', ()=> {
    const component = shallow(<RequestBatchCardModal />);

    expect(component.text().includes('Request Sent Successfully')).toBe(true);
});

test('component snapshot test', ()=>{
    const component = shallow(<RequestBatchCardModal/>);
    expect(component).toMatchSnapshot();
})
