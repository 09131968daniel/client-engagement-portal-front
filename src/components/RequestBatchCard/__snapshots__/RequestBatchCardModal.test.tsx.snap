// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshot test 1`] = `
<div
  className="RequestBatchCardModal"
>
  <div>
    Request Sent Successfully
  </div>
  <div
    className="XButton"
  >
    X
  </div>
</div>
`;
